common/core/desktop/src/kmx/kmx_xstring.cpp

```diff
@@ -173,43 +173,18 @@ PKMX_WCHAR km::kbp::kmx::decxstr(PKMX_WCHAR p, PKMX_WCHAR pStart)
   {
     return p-1;
   }
+
   else if(*(p-1) == UC_SENTINEL) return p-1;
-  else if(p > pStart+1 && *(p-2) == UC_SENTINEL)
-  {
-    switch(*(p-1))
-    {
-      case CODE_ANY:
-      case CODE_NOTANY:
-      case CODE_USE:
-      case CODE_DEADKEY:
-      case CODE_CLEARCONTEXT:
-      case CODE_CALL:
-      case CODE_CONTEXTEX:
-      case CODE_RESETOPT:
-      case CODE_SAVEOPT:
-        return p-2;
-    }
-  }
-  else if(p > pStart+2 && *(p-3) == UC_SENTINEL)
+  else
   {
-    switch(*(p-2))
+    for (int i = 1; i < 5; i++)
```

Comment:
I think we could start at `i = 2` since we checked `*(p-1)` above.

Reply:
One other thing: can we avoid the use of the magic number `5` here and instead define a constant `#define CODE__SIZE_MAX 5` alongside the `CODE__SIZE` definition?

Reply:
OK